### ISSUE TITLE
consider using std::priority_queue and remove PriorityQueue

### ISSUE THREAD

**Issue by facontidavide** [iris-ua/iris_lama#2]
I was curious so I took a look at your implementation of PriorityQueue.

I did some benchmark and  `std::priority_queue< std::pair<int,T>>` is actually faster (double as much).

Therefore, I believe it would be better to remove it, since the STL implementation is better and well supported.
You can see the results of the benchmark and the code here: http://quick-bench.com/SzpCCMx3aqGrAp-gjpuL6XvXEOg

**Comment by eupedrosa** [comment]
I tested your suggestion and there is _no_ noticeable difference in the overall execution times. This does not mean that `std::priority_queue` is not better. It means that the `PriorityQueue` is not a bottleneck.

Nonetheless, I like how the STL implementation is much cleaner. I will push it.

**Comment by facontidavide** [comment]
yes, I supposed it is not a bottleneck... just less code to maintain ;)

Remember that you need to implement the custom **operator <** that is missing in my example.
